spec/formula_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'spec_helper.rb'
+
+describe Statsample::GLM::Formula do
+  context '#parse_formula' do
+    context 'no interaction' do
+      let(:formula) { described_class.new 'y ~ a+b' }
+      subject { formula.parse_formula }
+      
+      it { is_expected.to be_a Array }
+      its(:first) { is_expected.to eq Statsample::GLM::Token }
+      its(:to_basic_formula) { is_expected.to eq 'y ~ 1+a(-)+b(-)' }
```

Comment:
I am not sure (because don't want to interrupt at early stages of design), but "intuitively" for me the next convention could possibly be:
* `Formula.new('y ~ ...')` creates **already parsed** formula;
* and the class `Formula` has methods like `#tokens`, `#to_basic_formula` and so on.

But that's from "bird view" look, I can miss everyting about it :)